### ISSUE TITLE
Users are able to "identify" items that should remain hidden

### ISSUE THREAD

**Issue by jchristian2** [foundryvtt/dnd5e#2781]
This was a feature that I was wanting prior to version 3.0.  The ability to add a magic item to a players inventory as a GM, but its true identity stays hidden until the GM changes the item to identified.  Now the option is there, but the player has the ability to identify the item themselves.  This should be a GM only option, and I am not seeing any settings in the config to change its visibility or permissions.

**Comment by gandb** [comment]
I totally agree with this. And the information need be hidden until identification, including the gp value.

**Comment by MaxPat931** [comment]
I wrote up this [world script](https://github.com/foundryvtt/dnd5e/wiki/Modifying-Your-Game-with-Scripts) that will hide the Identification buttons from a player when an item is Unidentified, so you can use this in the meantime while it gets scoped out by the system.

https://github.com/MaxPat931/Macros/blob/main/WorldScripts/HideUnidentify3.js 

gandb, the details and gp value are hidden when unidentified, the description will automatically change from the Description to the Unidentified Description for the player.

**Comment by voodoofrog** [comment]
I feel like this might be a good candidate for a new permission so that, say, trusted players are allowed to identify by themselves. That said, I'm not sure if that's purely in the gift of the core Foundry app itself.

**Comment by gandb** [comment]
> I feel like this might be a good candidate for a new permission so that, say, trusted players are allowed to identify by themselves. That said, I'm not sure if that's purely in the gift of the core Foundry app itself.

I tottaly agree with you, but a lot of permissions aren't includded into the core, can be added into the dnd5e

**Comment by zero127** [comment]
> I wrote up a [world script](https://github.com/foundryvtt/dnd5e/wiki/Modifying-Your-Game-with-Scripts) that will hide the Identification buttons from a player when an item is Unidentified, so you can use this in the meantime while it gets scoped out by the system.
> 
> https://github.com/MaxPat931/Macros/blob/main/WorldScripts/HideUnidentify3.js
> 
> gandb, the details and gp value are hidden when unidentified, the description will automatically change from the Description to the Unidentified Description for the player.

Bless you, you wonderful person. This works a treat.

**Comment by MaxPat931** [comment]
Could this fit into the "Configure Visibility" options implemented in #4798?

**Comment by gandb** [comment]
@MaxPat931  the problem isnt description in the chat, is in all parts and forbiden player identify item without permission of DM, like the hideunidentify3.js from @zero127

**Comment by MaxPat931** [comment]
I didnt say anything about chat, I just linked to the wrong issue initially since it is linked to the PR I meant to mention, Im asking that the option to remove the Identify button could be added into the new Configure Visibility setting group added in PR #4817 .

Your screenshots seems to be from a GM account, all of that information _is_ hidden from a player if the item is Unidentified 
![image](https://github.com/user-attachments/assets/fecb88f8-62ba-446d-a9f8-e832e0cf9d6d)
The description above is from the "Unidentified Description" which is only available to GMs (see Mace of Disruption in the SRD)
![image](https://github.com/user-attachments/assets/c048654f-e617-4fb7-910c-5c0a165bd58b)

There was a bug related to this that was fixed in 4.1.2 #4730, but that doesnt look like what you're showing

**Comment by gandb** [comment]
Oh, I understand now, sorry. Yes, but need block all things in the images

**Comment by Fmerrick** [comment]
@MaxPat931 Does your script for for v5 of the 5e system? I execute the macro but nothing appears to happen.

**Comment by neovatar** [comment]
@Fmerrick 

late to the party, but the following code worked for me with dnd5e 5.3.x

```
// Prevent Players from Identifying their items by hiding the buttons from them, only the GM will be able to Identify them
// Remove Identify button at top of Item Sheet
Hooks.on("renderItemSheet5e", (sheet, [html]) => {
  if (game.user.isGM) return;
  const unidentified = sheet.item.system.identified === false;
  if (!unidentified) return;
  html.parentElement.querySelectorAll('[data-action="toggleState"][data-property="system.identified"]').forEach(n => n.remove());
});

// Remove Identify button from Item Context menu on Actor Sheet
Hooks.on("dnd5e.getItemContextOptions", (item, buttons) => {
  if (game.user.isGM) return;
  const unidentified = item.system.identified === false;
  if (!unidentified) return;
  const removeUnid = buttons.findIndex(option => option.name === 'DND5E.Identify');
  buttons.findSplice(e => e.name === "DND5E.Identify");
});
```
